### DISCUSSION TITLE
Is there a way to use same theme for all grammars?

### DISCUSSION THREAD

**Discussion by iloveeclipse** [eclipse/tm4e#660] [General]
I personally do not really like "Solarized Light" theme and would be happy to use "Eclipse Light" for **all** grammars in TextMate, but it seem there is no way to change defaults in the UI **for all** grammars - one has to go through and change every single one to the "right" theme.

Is there some preference one can set to change "default theme" from "Solarized Light" to "Eclipse Light"?

**Comment by sebthom** [comment]
I don't think that is currently possible.

The default theme is defined via an extension point:
https://github.com/eclipse/tm4e/blob/854d55a638f15c49ea41a818a2aced73c44b0f67/org.eclipse.tm4e.ui/plugin.xml#L26-L31

and the ThemeManager iterates over all registered themes to find the default theme:
https://github.com/eclipse/tm4e/blob/854d55a638f15c49ea41a818a2aced73c44b0f67/org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/themes/AbstractThemeManager.java#L65-L68

When two themes are registered as default the first one registered will be returned.

There is no code in place to change or track the default theme per workspace.

PRs are welcome :-)

**Comment by iloveeclipse** [comment]
OK, so next question: can we have "Eclipse" the default theme for Eclipse?

**Reply by sebthom** [reply]
> OK, so next question: can we have "Eclipse" the default theme for Eclipse?

@mickaelistria @akurtakov @angelozerr WDYT?

**Reply by mickaelistria** [reply]
I personally now prefer the Solarized theme much better than the Eclipse one, so I don't really support the idea of changing the default theme.
I think we should instead consider having in TM4E a preference to decide the default theme, instead of relying on some hardcoded extension.

**Reply by sebthom** [reply]
> I personally now prefer the Solarized theme much better than the Eclipse one

Me too TBH.

**Reply by angelozerr** [reply]
It is just a feeling (and perhaps it is because I'm using vscode), but I found `Solarized theme` is more modern than Eclipse theme.

**Comment by sebthom** [comment]
> Is there some preference one can set to change "default theme" from "Solarized Light" to "Eclipse Light"?

You will be able to set the default theme in the TextMate UI settings with the next TM4E release, see https://github.com/eclipse/tm4e/pull/686

**Reply by iloveeclipse** [reply]
Great! Thanks.
